front/lib/api/assistant/legacy_agent.ts

```diff
@@ -376,15 +363,20 @@ async function* runAction(
   }
 
   if (isRetrievalConfiguration(action)) {
```

Comment:
Goal is still to get rid of this right? (Assuming we move the code handling the events and other custom cases.

Reply:
Yep last step is to move the streaming events to their respective classes.